File: package/umt_python/src/string/capitalize_word.py
def capitalize_word(word: str) -> str:
    """
    Uppercases the first character of a word and lowercases the rest.

    Args:
        word: Input word.

    Returns:
        The normalized word with its first character uppercased and the
        remaining characters lowercased.

    Example:
        >>> capitalize_word("hello")
        'Hello'
        >>> capitalize_word("hELLO")
        'Hello'
        >>> capitalize_word("")
        ''
        >>> capitalize_word("a")
        'A'
    """
    return word[:1].upper() + word[1:].lower()

File: package/umt_rust/src/string/capitalize_word.rs
/// Capitalizes the first character of a word and lowercases the rest.
///
/// # Arguments
/// * `word` - The word to transform
///
/// # Returns
/// The word with its first character uppercased and the remainder lowercased
///
/// # Examples
/// ```
/// use umt_rust::string::umt_capitalize_word;
/// assert_eq!(umt_capitalize_word("hello"), "Hello");
/// assert_eq!(umt_capitalize_word("hELLO"), "Hello");
/// assert_eq!(umt_capitalize_word(""), "");
/// ```
#[inline]
pub fn umt_capitalize_word(word: &str) -> String {
    let mut chars = word.chars();
    match chars.next() {
        Some(first) => first.to_uppercase().collect::<String>() + &chars.as_str().to_lowercase(),
        None => String::new(),
    }
}
